### PR TITLE
DOC: Make section title capitalization consistent #26830

### LINKED ISSUE #26830: DOC: Make section title capitalization consistent
At the moment we've got a mix of capitalization formats in the title sections of our docs. For example, in the first level of the TOC tree in http://dev.pandas.io/ we've got (not the second word first letter, `s` lower case vs `G` upper case):
- `Getting started`
- `User Guide`

And this happens all across the documentation. I think it makes more sense to just capitalize the first letter of the first word, like in `Getting started`. More a personal preference than any objective reason, but looks simpler, more standard, and we avoid things like `pandas Ecosystem` which doesn't look really great to me.

CC: @TomAugspurger @jorisvandenbossche

### PR DESCRIPTION
**Pull request by restlessadmin**
- [x] closes #26830
Adjusted titles in Docs to have consistent first letter only capitalization, unless proper noun in title.

### CODE REVIEW

File: doc/source/development/contributing.rst [hunk 1]
```diff
@@ -605,7 +605,7 @@ and run ``flake8`` on them, one after the other.
 
 .. _contributing.import-formatting:
 
-Import Formatting
+Import formatting
```
Comment (jreback) [inline doc/source/development/contributing.rst]: not sure why should change section headers like this

File: doc/source/development/extending.rst [hunk 2]
```diff
@@ -3,7 +3,7 @@
 {{ header }}
 
 ****************
-Extending Pandas
+Extending pandas
```
Comment (jreback) [inline doc/source/development/extending.rst]: shouldn't change major headings

**Review by jreback — CHANGES_REQUESTED** [review]
I believe most of these are actually purposful, what was the intent on the changes here? e.g. section headers should be more sentence like?

cc @jorisvandenbossche @datapythonista

**Comment by jreback** [comment]
@datapythonista @jorisvandenbossche I see your discussion, but do we really want this to apply to *all* headers; what is the theory here?

**Comment by pmaxey83** [comment]
I think I went too literal on all the headers.  I will cancel this PR.

**Comment by jorisvandenbossche** [comment]
@jreback Let's then have the discussion on the issue. Can you comment there what logic you would follow on which headers to apply and which not? (our theory: consistency for all headers)

> I think I went too literal on all the headers. I will cancel this PR.

@pmaxey83 no, no, you did what was described in the issue. It's just that we don't yet fully agree about that. You will need to have some patience.

**Comment by pmaxey83** [comment]
Oh, I am sorry. I thought I did the change incorrectly.

**Comment by jorisvandenbossche** [comment]
As I said, no need to be sorry. As you did what was described correctly. There is only not yet a full agreement on what was described there.

**Comment by jorisvandenbossche** [comment]
@pmaxey83 Can you reopen the pull request, or restore your branch? (depending on the outcome of the discussion, those changes are useful)

**Comment by datapythonista** [comment]
Thanks @pmaxey83, great job here. Sorry it's been my fault that I assumed the discussion in this issue was finished too early. Also, I wasn't clear, but my idea was to start only with the title sections visible in the home page, and take care of the rest later.

As @jorisvandenbossche said, please restore your remote branch, there is a lot of great work there I hope we want to get into the docs.

@jreback my points to defend this:
- Consistency across all documentation headers
- It's much easier to keep consistency if the standard is simpler (and title case is somehow ambiguous if you're not a linguist I'd say `Implement one Thing` vs `Implement One Thing`)
- I understand that title case gives some emphasis, but the documentation renders titles in bigger letters and bold, so doesn't seem that needed to me
- Given that we prefer `pandas` in lowercase, title case makes some titles quite ugly in my opinion `pandas Ecosystem` is the main one (in the top level).

What is the main advantage you see to use title case?

**Comment by jorisvandenbossche** [comment]
Marc, let's have the discussion on the issue. Can you copy your arguments there?

**Comment by jreback** [comment]
@pmaxey83 yeah didn't mean for you to close this, I had already commented when I actually read the issue :->

File: doc/source/getting_started/basics.rst [hunk 3]
```diff
@@ -643,7 +643,7 @@ there for details about accepted inputs.
 
 .. _basics.idxmin:
 
-Index of Min/Max Values
+Index of min/max Values
```
Comment (datapythonista) [inline doc/source/getting_started/basics.rst]: Values also

File: doc/source/user_guide/categorical.rst [hunk 4]
```diff
@@ -1191,7 +1191,7 @@ change the original ``Categorical``:
     df["cat"].cat.categories = [1, 2, 3, 4, 5]
     cat
 
-Use ``copy=True`` to prevent such a behaviour or simply don't reuse ``Categoricals``:
+Use ``copy=True`` to prevent such a behavior or simply don't reuse ``Categoricals``:
```
Comment (datapythonista) [inline doc/source/user_guide/categorical.rst]: haha, you're starting a British vs American English war here, but good catch ;)

File: doc/source/user_guide/enhancingperf.rst [hunk 5]
```diff
@@ -15,7 +15,7 @@ when we use Cython and Numba on a test function operating row-wise on the
 
 .. _enhancingperf.cython:
 
-Cython (Writing C extensions for pandas)
+Cython (writing C extensions for Pandas)
```
Comment (datapythonista) [inline doc/source/user_guide/enhancingperf.rst]: We use `pandas` lowercase, as a trademark, can you restore this one please

File: doc/source/user_guide/gotchas.rst [hunk 6]
```diff
@@ -75,10 +75,10 @@ See also :ref:`Categorical Memory Usage <categorical.memory>`.
 
 .. _gotchas.truth:
 
-Using If/Truth Statements with pandas
+Using if/truth statements with Pandas
```
Comment (datapythonista) [inline doc/source/user_guide/gotchas.rst]: same regarding pandas

File: doc/source/user_guide/gotchas.rst [hunk 7]
```diff
@@ -75,10 +75,10 @@ See also :ref:`Categorical Memory Usage <categorical.memory>`.
...
 -------------------------------------
 
-pandas follows the NumPy convention of raising an error when you try to convert
+Pandas follows the NumPy convention of raising an error when you try to convert
```
Comment (datapythonista) [inline doc/source/user_guide/gotchas.rst]: same

File: doc/source/user_guide/groupby.rst [hunk 8]
```diff
@@ -146,7 +146,7 @@ columns:
 
     In [5]: grouped = df.groupby(get_letter_type, axis=1)
 
-pandas :class:`~pandas.Index` objects support duplicate values. If a
+Pandas :class:`~pandas.Index` objects support duplicate values. If a
```
Comment (datapythonista) [inline doc/source/user_guide/groupby.rst]: same

File: doc/source/user_guide/groupby.rst [hunk 9]
```diff
@@ -1422,7 +1422,7 @@ In the following examples, **df.index // 5** returns a binary array which is use
    df.index // 5
    df.groupby(df.index // 5).std()
 
-Returning a Series to propagate names
+Returning a series to propagate names
```
Comment (datapythonista) [inline doc/source/user_guide/groupby.rst]: `Series` is the name of the class, you can leave it capitalized

File: doc/source/user_guide/io.rst [hunk 10]
```diff
@@ -13,11 +13,11 @@
...
 
 ===============================
-IO Tools (Text, CSV, HDF5, ...)
+IO tools (text, CSV, HDF5, ...)
 ===============================
 
-The pandas I/O API is a set of top level ``reader`` functions accessed like
-:func:`pandas.read_csv` that generally return a pandas object. The corresponding
+The Pandas I/O API is a set of top level ``reader`` functions accessed like
+:func:`pandas.read_csv` that generally return a Pandas object. The corresponding
```
Comment (datapythonista) [inline doc/source/user_guide/io.rst]: same about pandas

File: doc/source/user_guide/io.rst [hunk 11]
```diff
@@ -1218,7 +1218,7 @@ Infinity
 These will ignore the case of the value, meaning ``Inf``, will also be parsed as ``np.inf``.
 
 
-Returning Series
+Returning series
```
Comment (datapythonista) [inline doc/source/user_guide/io.rst]: same about `Series`

File: doc/source/user_guide/options.rst [hunk 12]
```diff
@@ -120,10 +120,10 @@ are restored automatically when you exit the `with` block:
    print(pd.get_option("display.max_columns"))
 
 
-Setting Startup Options in python/ipython Environment
+Setting startup options in Python/ipython environment
 -----------------------------------------------------
 
-Using startup scripts for the python/ipython environment to import pandas and set options makes working with pandas more efficient.  To do this, create a .py or .ipy script in the startup directory of the desired profile.  An example where the startup folder is in a default ipython profile can be found at:
+Using startup scripts for the Python/ipython environment to import pandas and set options makes working with pandas more efficient.  To do this, create a .py or .ipy script in the startup directory of the desired profile.  An example where the startup folder is in a default ipython profile can be found at:
```
Comment (datapythonista) [inline doc/source/user_guide/options.rst]: Can you also capitalize `IPython` please (in the title and in the paragraph

File: doc/source/user_guide/timeseries.rst [hunk 13]
```diff
@@ -2007,7 +2007,7 @@ As with ``DatetimeIndex``, the endpoints will be included in the result. The exa
 
    dfp['2013-01-01 10H':'2013-01-01 11H']
 
-Frequency Conversion and Resampling with PeriodIndex
+Frequency conversion and resampling with periodIndex
```
Comment (datapythonista) [inline doc/source/user_guide/timeseries.rst]: restore `PeriodIndex` please, it's the name of a class

File: doc/source/user_guide/timeseries.rst [hunk 14]
```diff
@@ -2156,18 +2156,18 @@ These can easily be converted to a ``PeriodIndex``:
 
 .. _timeseries.timezone:
 
-Time Zone Handling
+Time zone handling
 ------------------
 
-pandas provides rich support for working with timestamps in different time
+Pandas provides rich support for working with timestamps in different time
```
Comment (datapythonista) [inline doc/source/user_guide/timeseries.rst]: `pandas` lowercase

File: doc/source/user_guide/timeseries.rst [hunk 15]
```diff
@@ -2156,18 +2156,18 @@ These can easily be converted to a ``PeriodIndex``:
...
 zones using the ``pytz`` and ``dateutil`` libraries or class:`datetime.timezone`
 objects from the standard library.
 
 
-Working with Time Zones
+Working with time zones
 ~~~~~~~~~~~~~~~~~~~~~~~
 
-By default, pandas objects are time zone unaware:
+By default, Pandas objects are time zone unaware:
```
Comment (datapythonista) [inline doc/source/user_guide/timeseries.rst]: same

File: doc/source/whatsnew/v0.15.0.rst [hunk 16]
```diff
@@ -412,7 +412,7 @@ Rolling/Expanding Moments improvements
 
     s = pd.Series([10.5, 8.8, 11.4, 9.7, 9.3])
 
-  Behavior prior to 0.15.0:
+  behavior prior to 0.15.0:
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.15.0.rst]: not sure about this one, I guess you have a reason, but looks correct as it was.

File: doc/source/whatsnew/v0.15.0.rst [hunk 17]
```diff
@@ -770,7 +770,7 @@ Other notable API changes:
      s2 = s
      s += 1.5
 
-  Behavior prior to v0.15.0
+  behavior prior to v0.15.0
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.15.0.rst]: same

File: doc/source/whatsnew/v0.17.0.rst [hunk 18]
```diff
@@ -681,15 +681,15 @@ New Behavior:
       pd.Timestamp.now()
       pd.Timestamp.now() + offsets.DateOffset(years=1)
 
-Changes to Index Comparisons
+Changes to index comparisons
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.17.0.rst]: `Index` is the class, you can leave it with capital `I`

File: doc/source/whatsnew/v0.17.0.rst [hunk 19]
```diff
@@ -729,7 +729,7 @@ or it can return False if broadcasting can not be done:
 
    np.array([1, 2, 3]) == np.array([1, 2])
 
-Changes to Boolean Comparisons vs. None
+Changes to Boolean comparisons vs. none
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.17.0.rst]: I'd probably leave `None` as in Python, and may be `boolean`, even if I see your point. Up to you, just a personal preference here.

File: doc/source/whatsnew/v0.17.0.rst [hunk 20]
```diff
@@ -917,7 +917,7 @@ Other API Changes
 - ``NaT``'s methods now either raise ``ValueError``, or return ``np.nan`` or ``NaT`` (:issue:`9513`)
 
   ===============================     ===============================================================
-  Behavior                            Methods
+  behavior                            Methods
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.17.0.rst]: This is the header of the table, it was good before.

File: doc/source/whatsnew/v0.19.0.rst [hunk 21]
```diff
@@ -335,7 +335,7 @@ Using the anchoring suffix, you can also specify the day of month to use instead
 
 .. _whatsnew_0190.enhancements.index:
 
-New Index methods
+New index methods
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.19.0.rst]: same, class name here, leave `Index`

File: doc/source/whatsnew/v0.20.0.rst [hunk 22]
```diff
@@ -568,7 +568,7 @@ then write them out again after applying the procedure below.
 
 .. _whatsnew_0200.api_breaking.index_map:
 
-Map on Index types now return other Index types
+Map on index types now return other index types
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.20.0.rst]: same about `Index`

File: doc/source/whatsnew/v0.20.0.rst [hunk 23]
```diff
@@ -823,12 +823,12 @@ currently released version of ``pandas-gbq=0.1.4``. Documentation is now hosted
 
 .. _whatsnew_0200.api_breaking.memory_usage:
 
-Memory Usage for Index is more Accurate
+Memory usage for index is more accurate
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.20.0.rst]: again `Index`

File: doc/source/whatsnew/v0.20.0.rst [hunk 24]
```diff
@@ -1352,7 +1352,7 @@ Using ``.iloc``. Here we will get the location of the 'A' column, then use *posi
 
 .. _whatsnew_0200.api_breaking.deprecate_panel:
 
-Deprecate Panel
+Deprecate panel
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.20.0.rst]: `Panel` is also a class name

File: doc/source/whatsnew/v0.23.0.rst [hunk 25]
```diff
@@ -186,7 +186,7 @@ resetting indexes. See the :ref:`Sorting by Indexes and Values
 
 .. _whatsnew_023.enhancements.extension:
 
-Extending Pandas with Custom Types (Experimental)
+Extending Pandas with custom types (experimental)
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.23.0.rst]: lowercase `pandas`

File: doc/source/whatsnew/v0.23.0.rst [hunk 26]
```diff
@@ -639,7 +639,7 @@ If you wish to retain the old behavior while using Python >= 3.6, you can use
 
 .. _whatsnew_0230.api_breaking.deprecate_panel:
 
-Deprecate Panel
+Deprecate panel
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.23.0.rst]: same about `Panel`

File: doc/source/whatsnew/v0.23.1.rst [hunk 27]
```diff
@@ -20,10 +20,10 @@ and bug fixes. We recommend that all users upgrade to this version.
 
 .. _whatsnew_0231.fixed_regressions:
 
-Fixed Regressions
+Fixed regressions
 ~~~~~~~~~~~~~~~~~
 
-**Comparing Series with datetime.date**
+**Comparing series with datetime.date**
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.23.1.rst]: `Series` with capital `S`, it's a class

File: doc/source/whatsnew/v0.24.0.rst [hunk 28]
```diff
@@ -105,7 +105,7 @@ See :ref:`integer_na` for more.
 
 .. _whatsnew_0240.values_api:
 
-Accessing the values in a Series or Index
+Accessing the values in a series or index
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.24.0.rst]: same about class names

File: doc/source/whatsnew/v0.24.0.rst [hunk 29]
```diff
@@ -194,7 +194,7 @@ is a float.
 
 .. _whatsnew_0240.enhancements.interval:
 
-Storing Interval and Period Data in Series and DataFrame
+Storing interval and period data in series and DataFrame
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.24.0.rst]: `Interval` and `Period` are classes too, also `Series`

File: doc/source/whatsnew/v0.24.0.rst [hunk 30]
```diff
@@ -1105,13 +1105,13 @@ broadcast.  (:issue:`23000`)
 
 .. _whatsnew_0240.api.incompatibilities:
 
-Series and Index Data-Dtype Incompatibilities
+Series and index data-dtype incompatibilities
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.24.0.rst]: `Index` is a class

File: doc/source/whatsnew/v0.24.0.rst [hunk 31]
```diff
@@ -1417,7 +1417,7 @@ The old behavior can be retained with by localizing directly to the final timezo
 
 .. _whatsnew_0240.deprecations.tz_aware_array:
 
-Converting Timezone-Aware Series and Index to NumPy Arrays
+Converting timezone-aware series and index to NumPy arrays
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.24.0.rst]: `Series` and `Index`, class names

File: doc/source/whatsnew/v0.25.0.rst [hunk 32]
```diff
@@ -286,7 +286,7 @@ of ``object`` dtype. :attr:`Series.str` will now infer the dtype data *within* t
 
 .. _whatsnew_0250.api_breaking.incompatible_index_unions:
 
-Incompatible Index Type Unions
+Incompatible index type unions
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.25.0.rst]: same about `Index`

File: doc/source/whatsnew/v0.7.0.rst [hunk 33]
```diff
@@ -227,7 +227,7 @@ If the index had been sorted, the "range selection" would have been possible:
    g   -1.182230
    dtype: float64
 
-Changes to Series ``[]`` operator
+Changes to series ``[]`` operator
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.7.0.rst]: same

**Review by datapythonista — COMMENTED** [review]
Amazing clean up @pmaxey83, thanks a lot for the effort.

We're moving forward with this, just few cases that should be changed (mainly class names that we want to keep with the first letter capitalized), and we're ready to merge.

**Comment by pmaxey83** [comment]
Once I'm home from work, I will adjust the keywords.

Thank you.

**Comment by datapythonista** [comment]
Since your remote branch has been deleted, may be it's better to rename your local branch `git branch -m <new_name>` from your branch, push to a new remote branch, and open a new PR (referencing this to keep them linked). I guess GitHub is smart enough to keep everything (code history, comments,...), but not sure.

**Comment by pmaxey83** [comment]
I had a few issues when attempting to push to my remote branch, so I
decided it might be more clean to backup my changes and clone a fresh copy
of the repository from pandas-dev.  Beyond that, the above corrections have
been made.

**Comment by datapythonista** [comment]
I was guessing that, that's why I suggested to rename the branch and open a new PR. I see you opened #26830. For the next time, it's usually a good practice to reference this PR there, so it's easier to go from one to the other, and in the history we know what happened, in case we need to investigate problems.

**Comment by pmaxey83** [comment]
Sorry about that. I'm still a bit new to Git and was struggling a bit to
get the push working correctly.  I will keep that in mind for any future
changes.  Thank you for your patience, I really appreciate it.

**Comment by datapythonista** [comment]
no worries, nobody does it the first time, that's why I was telling you how we usually do it :)

**Pull request closed without merging**